### PR TITLE
Migrating module to fabric

### PR DESCRIPTION
**Pull request by SudoPlz**
# Overview

Making changes to make sure this library supports fabric.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,6 +37,18 @@ The package is [automatically linked](https://github.com/react-native-community/
 npx pod-install
 ```
 
+## Fabric
```
Comment (Naturalclar) [inline README.md:40]: @SudoPlz 
I believe this part of the readme is not necessary as the podfile automatically includes those headers when ` :fabric_enabled(:fabric_enabled => true)`. What do you think?

File: src/types.ts [hunk 2]
```diff
@@ -149,5 +150,6 @@ export type NativeMenuComponentProps = {
   style?: StyleProp<ViewStyle>;
   onPressAction?: ({ nativeEvent }: NativeActionEvent) => void;
   actions: ProcessedMenuAction[];
+  actionsHash: string;
```
Comment (Naturalclar) [inline src/types.ts:153]: [question] is this change required for the fabric architecture? if not, I'd like to keep it separate from this PR

**Review by Naturalclar — CHANGES_REQUESTED** [review]
@SudoPlz Thank you for the pull request! I've made some comments about the PR.

File: README.md (hunk 1 above)
Comment (SudoPlz) [reply README.md:40]: that might be true, we need to double check. On my project I actually needed to add those to my Podfile because the `modular_headers` option wasn't added automatically.

For other projects this may not be required, but I guess it depends on how the Podfile of the consuming iOS project is structured.

**Review by Naturalclar — APPROVED** [review]
LGTM, thanks!

**Comment by alexanderblackh** [comment]
For whatever it's worth, I can confirm running `yarn add react-native-menu/menu#master` this does work on React Native 73.6 with Fabric on Android.  Hope to see this packaged in a release soon.
